src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -304,9 +304,14 @@ private void WriteMvidsTo(Metadata metadata, ObjectWriter writer, CancellationTo
 
             if (metadata is AssemblyMetadata assemblyMetadata)
             {
+                if (!TryGetModules(assemblyMetadata, out var modules))
+                {
+                    // Gracefully bail out without writing anything to the writer.
```

Comment:
Is `writer.WriteInt32(MetadataFailed);` needed here?

Reply:
I am matching the implementation for other bail out case in this method.